code/datums/action.dm

```diff
@@ -0,0 +1,406 @@
+
+		ApplyIcon(button)
+
+		if(!IsAvailable())
+			button.color = rgb(128,0,0,128)
+		else
+			button.color = rgb(255,255,255,255)
+			return 1
+
+/datum/action/proc/ApplyIcon(obj/screen/movable/action_button/current_button)
+	current_button.overlays.Cut()
+	if(icon_icon && button_icon_state)
+		var/image/img
+		img = image(icon_icon, current_button, button_icon_state)
+		img.pixel_x = 0
+		img.pixel_y = 0
+		current_button.overlays += img
+
+//Presets for item actions
+/datum/action/item_action
+	check_flags = AB_CHECK_RESTRAINED|AB_CHECK_STUNNED|AB_CHECK_LYING|AB_CHECK_CONSCIOUS
+
+/datum/action/item_action/New(Target)
+	..()
+	var/obj/item/I = target
+	I.actions += src
+
+/datum/action/item_action/Destroy()
+	var/obj/item/I = target
+	I.actions -= src
+	return ..()
+
+/datum/action/item_action/Trigger()
+	if(!..())
+		return 0
+	if(target)
+		var/obj/item/I = target
+		I.ui_action_click(owner, type)
+	return 1
+
+/datum/action/item_action/ApplyIcon(obj/screen/movable/action_button/current_button)
+	current_button.overlays.Cut()
+	if(target)
+		var/obj/item/I = target
+		var/old_layer = I.layer
+		var/old_plane = I.plane
+		I.layer = 21
+		I.plane = HUD_PLANE
+		current_button.overlays += I
+		I.layer = old_layer
+		I.plane = old_plane
+
+/datum/action/item_action/toggle_light
+	name = "Toggle Light"
+
+/datum/action/item_action/toggle_hood
+	name = "Toggle Hood"
+
+/datum/action/item_action/toggle_firemode
+	name = "Toggle Firemode"
+
+/datum/action/item_action/startchainsaw
+	name = "Pull The Starting Cord"
+
+/datum/action/item_action/toggle_gunlight
+	name = "Toggle Gunlight"
+
+/datum/action/item_action/toggle_mode
+	name = "Toggle Mode"
+
+/datum/action/item_action/toggle_barrier_spread
+	name = "Toggle Barrier Spread"
+
+/datum/action/item_action/equip_unequip_TED_Gun
+	name = "Equip/Unequip TED Gun"
+
+/datum/action/item_action/toggle_paddles
+	name = "Toggle Paddles"
+
+/datum/action/item_action/set_internals
+	name = "Set Internals"
+
+/datum/action/item_action/set_internals/UpdateButtonIcon()
+	if(..()) //button available
+		if(iscarbon(owner))
+			var/mob/living/carbon/C = owner
+			if(target == C.internal)
+				button.icon_state = "bg_default_on"
+
+/datum/action/item_action/toggle_mister
+	name = "Toggle Mister"
+
+/datum/action/item_action/toggle_helmet_light
+	name = "Toggle Helmet Light"
+
+/datum/action/item_action/toggle_helmet_mode
+	name = "Toggle Helmet Mode"
+
+/datum/action/item_action/toggle_hardsuit_mode
+	name = "Toggle Hardsuit Mode"
+
+/datum/action/item_action/toggle
+/datum/action/item_action/toggle/New(Target)
+	..()
+	name = "Toggle [target.name]"
+	button.name = name
+
+/datum/action/item_action/openclose
+/datum/action/item_action/openclose/New(Target)
+	..()
+	name = "Open/Close [target.name]"
+	button.name = name
+
+/datum/action/item_action/button
+/datum/action/item_action/button/New(Target)
+	..()
+	name = "Button/Unbutton [target.name]"
+	button.name = name
+
+/datum/action/item_action/zipper
```

Comment:
I'd rather not, they have no purpose right now. It looks better to have them close.